backends/kv/reader.go

```diff
@@ -170,12 +178,15 @@ func (ki *Iterator) Next() bool {
 			}
 
 			var err error
-			err = utils.AppendNil(col)
+			err = utils.AppendNil(byName[name])
 			if err != nil {
 				ki.err = err
 				return false
 			}
+			currentNullMask.NullColumns[name] = true
```

Comment:
IMHO it's better to use integer index where possible. In case your data structure has consistent column index - use integer index instead of name for faster access and memory efficiency